SYCL/ESIMD/regression/math_const_fix_test.cpp

```diff
@@ -0,0 +1,229 @@
+
+  shared_vector<float> vector_output_atan2_fast(SIMD, allocator);
+  shared_vector<float> vector_output_atan2(SIMD, allocator);
+  shared_vector<float> scalar_output_atan2_fast(1, allocator);
+  shared_vector<float> scalar_output_atan2(1, allocator);
+
+  float ha = x;
+  float hb = y;
+  {
+
+    auto e = queue.submit([&](sycl::handler &cgh) {
+      float *vector_output_atan2_fast_ptr = vector_output_atan2_fast.data();
+      float *scalar_output_atan2_fast_ptr = scalar_output_atan2_fast.data();
+      float *vector_output_atan2_ptr = vector_output_atan2.data();
+      float *scalar_output_atan2_ptr = scalar_output_atan2.data();
+
+      auto kernel = ([=]() [[intel::sycl_explicit_simd]] {
+        using namespace sycl::ext::intel::esimd;
+
+        simd<float, SIMD> a = ha;
+        simd<float, SIMD> b = hb;
+
+        simd<float, SIMD> vector_result =
+            sycl::ext::intel::experimental::esimd::atan2_fast(a, b);
+        simd<float, 1> scalar_result =
+            sycl::ext::intel::experimental::esimd::atan2_fast(ha, hb);
+        simd<float, SIMD> vector_result1 =
+            sycl::ext::intel::experimental::esimd::atan2(a, b);
+        simd<float, 1> scalar_result1 =
+            sycl::ext::intel::experimental::esimd::atan2(ha, hb);
+
+        vector_result.copy_to(vector_output_atan2_fast_ptr);
+        scalar_result.copy_to(scalar_output_atan2_fast_ptr);
+        vector_result1.copy_to(vector_output_atan2_ptr);
+        scalar_result1.copy_to(scalar_output_atan2_ptr);
+      });
+
+      cgh.single_task<class TestAtan>(kernel);
+    });
+    queue.wait();
+  }
+
+  float std_result_atan = std::atan2(ha, hb);
+
+  if (std::fabs(std_result_atan - scalar_output_atan2[0]) > 0.0001f) {
+    if (ErrCnt++ < 100)
+      std::cout << "Scalar test failed for atan2 for " << x << "," << y << "."
+                << std::endl;
+    return 1;
+  }
+
+  if (std::fabs(std_result_atan - scalar_output_atan2_fast[0]) > 0.1f) {
+    if (ErrCnt++ < 100)
+      std::cout << "Scalar test failed for atan2_fast for " << x << "," << y
+                << "." << std::endl;
+    return 1;
+  }
+
+  for (int i = 0; i < SIMD; ++i) {
+    if (scalar_output_atan2[0] != vector_output_atan2[i]) {
+      if (ErrCnt++ < 100)
+        std::cout << "Vector test failed for atan2 " << x << "," << y << "."
+                  << std::endl;
+      return 1;
+    }
+
+    if (scalar_output_atan2_fast[0] != vector_output_atan2_fast[i]) {
+      if (ErrCnt++ < 100)
+        std::cout << "Vector test failed for atan2_fast for " << x << "," << y
+                  << "." << std::endl;
+      return 1;
+    }
+  }
+
+  return 0;
+}
+
+int main() {
+
+  int test_result = 0;
+
+  for (float x = -10.f; x < 10.f; x += 0.25f) {
+    test_result |= test_sin_cos_atan(x);
+  }
+
+  for (float x = -5.f; x < 5.f; x += 0.25f) {
+    for (float y = -5.f; y < 5.f; y += 0.25f) {
+      if (std::abs(x) > 0.00001f && std::abs(y) > 0.00001f)
+        test_result |= test_atan2(x, y);

```

Comment:
BTW, Previously, the test failed with timeout. I think I see what is happening here.
The kernel runs about 1600 times, right? Scheduling/running a kernel takes time.

If you want to make this test working really fast, then you can create a kernel being run only once; it may have range<2>{40,40} and every of work-items would compute atan for (i/0.25, j/0.25) value and storing to some longer vector/mem.